### PR TITLE
M701 and M702 support from LCD and Gcode

### PR DESCRIPTION
**Pull request by q3ok**
From the M600 I've created the M701 and M702 support, and I've added it as new menu items to Prepare menu. There is no translations, its only in en language for display. The filament load are using still the screen from M600 with "Resume print" message, it has to be changed also.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -7318,6 +7320,143 @@ inline void gcode_M503() {
...
+    lcd_filament_change_show_message(FILAMENT_CHANGE_MESSAGE_STATUS);
+	
+  }
+
+  /**
+   * M702: Unload filament
+   */
+  inline void gcode_M702() {
+    
+	  if (thermalManager.tooColdToExtrude(active_extruder)) {
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Please do a global search and replace to replace tabs with spaces.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -7318,6 +7320,143 @@ inline void gcode_M503() {
     lcd_filament_change_show_message(FILAMENT_CHANGE_MESSAGE_STATUS);
   }
 
+    /**
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Remove extra leading space(s) here.

**Comment by thinkyhead** [comment]
Overall the implementation looks good (though I have yet to review it fully). Note that all new features must be made optional and disabled by default, so you'll need to add a new configuration option to make it possible to enable/disable this feature, and set it disabled.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
Some spacing changes requested…

File: Marlin/Marlin_main.cpp (hunk 1 above)
Comment (q3ok) [reply Marlin/Marlin_main.cpp]: Global tabs has been replaced with spaces, I believe that i didn't missed any

File: Marlin/Marlin_main.cpp (hunk 2 above)
Comment (q3ok) [reply Marlin/Marlin_main.cpp]: Spaces removed

**Comment by q3ok** [comment]
The option FILAMENT_LOADUNLOAD_SUPPORT has been added in Configuration_adv

**Comment by tcm0116** [comment]
@q3ok I recently submitted a pull request (https://github.com/MarlinFirmware/Marlin/pull/6407) that deals with the Filament Reload and Pause features. We might want to coordinate since this one is related to that group.

**Comment by q3ok** [comment]
@tcm0116 I am not sure if even the pause feature should be unified with filament change. IMO that should be two different options, but I can agree that M600 can be splitted into at least two functions, as well to use with M701, M702 and M125.
After few days of using the M701 and M702 made from M600 i found that there is a bug after filament load, which prevents to start any print - I didn't found yet the reason.

**Comment by tcm0116** [comment]
@q3ok By unify, I meant that it would make sense to have them share a set of common functions in the code, but still use the respective M codes. This would prevent having very similar code in several different places that has had small changes made to it.

**Comment by tcm0116** [comment]
@q3ok In my setup, I'm using an E3D Cyclops hotend, and they actually recommend feeding both filaments cold as far as possible before heating up the nozzle. As such, it would be nice to have the option to load the filament with the nozzle cold, and possibly even the option of feeding both extruders simultaneously. This would then require that the hotend be heated up and then the filament loaded the rest of the way. Thoughts on this?

**Review by thinkyhead — APPROVED** [review]

**Comment by thinkyhead** [comment]
#7107 supplants this PR…

**Pull request closed without merging**